### PR TITLE
Vision v1p1beta1 samples

### PR DESCRIPTION
**Pull request by nnegrey**

### CODE REVIEW

File: vision/beta/cloud-client/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+### Analyze an image
+
+```
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/your-project-credentials.json
+```
+
+```
+java -cp target/vision-detect-beta-1.0.0-jar-with-dependencies.jar \
```
Comment (jabubake) [inline vision/beta/cloud-client/README.md]: You mention the exec:java option and then build a fat jar here : looks inconsistent.

File: vision/beta/cloud-client/pom.xml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  <dependencies>
+    <!-- [START dependencies] -->
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-vision</artifactId>
+      <version>1.14.0</version>
+    </dependency>
+    <!-- [END dependencies] -->
+
+    <dependency>
```
Comment (jabubake) [inline vision/beta/cloud-client/pom.xml]: Can you confirm this dependency is required ? I hope not, if it is, please file a bug on google-cloud-java

File: vision/beta/cloud-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      <groupId>com.google.auth</groupId>
+      <artifactId>google-auth-library-oauth2-http</artifactId>
+      <version>0.9.0</version>
+    </dependency>
+    <dependency>
```
Comment (jabubake) [inline vision/beta/cloud-client/pom.xml]: Is there a reason guava is included ?

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 4]
```diff
@@ -0,0 +1,1271 @@
...
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Detect {
+
+  /**
+   * Detects entities,sentiment and syntax in a document using the Natural Language API.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: Vision API

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 5]
```diff
@@ -0,0 +1,1271 @@
...
+              + "| web | web-entities | web-entities-include-geo | crop \n"
+              + "Path:\n\tA file path (ex: ./resources/wakeupcat.jpg) or a URI for a Cloud Storage "
+              + "resource (gs://...)\n",
+          Detect.class.getCanonicalName());
+      return;
+    }
+    String command = args[0];
+    String path = args.length > 1 ? args[1] : "";
+
+    if (command.equals("faces")) {
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: Worth calling out these options in the README with an example for each option.

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 6]
```diff
@@ -0,0 +1,1271 @@
...
+              annotation.getBoundingPoly());
+        }
+      }
+    }
+  }
+
+  /**
+   * Detects faces in the specified remote image.
+   *
+   * @param gcsPath The path to the remote file to perform face detection on.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: remote file on Google Cloud Storage

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 7]
```diff
@@ -0,0 +1,1271 @@
...
+        // For full list of available annotations, see http://g.co/cloud/vision/docs
+        for (EntityAnnotation annotation : res.getLabelAnnotationsList()) {
+          annotation.getAllFields().forEach((k, v) -> out.printf("%s : %s\n", k, v.toString()));
+        }
+      }
+    }
+  }
+
+  /**
+   * Detects labels in the specified remote image.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: image on Google Cloud Storage

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 8]
```diff
@@ -0,0 +1,1271 @@
...
+          out.printf("Landmark: %s\n %s\n", annotation.getDescription(), info.getLatLng());
+        }
+      }
+    }
+  }
+
+  /**
+   * Detects landmarks in the specified URI.
+   *
+   * @param url The path to the file to perform landmark detection on.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: url => uri

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 9]
```diff
@@ -0,0 +1,1271 @@
...
+        for (EntityAnnotation annotation : res.getLandmarkAnnotationsList()) {
+          LocationInfo info = annotation.getLocationsList().listIterator().next();
+          out.printf("Landmark: %s\n %s\n", annotation.getDescription(), info.getLatLng());
+        }
+      }
+    }
+  }
+
+  /**
+   * Detects landmarks in the specified remote image.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: image on Google Cloud Storage

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 10]
```diff
@@ -0,0 +1,1271 @@
...
+        // For full list of available annotations, see http://g.co/cloud/vision/docs
+        for (EntityAnnotation annotation : res.getLogoAnnotationsList()) {
+          out.println(annotation.getDescription());
+        }
+      }
+    }
+  }
+
+  /**
+   * Detects logos in the specified remote image.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: image on Google Cloud Storage

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 11]
```diff
@@ -0,0 +1,1271 @@
...
+        for (EntityAnnotation annotation : res.getTextAnnotationsList()) {
+          out.printf("Text: %s\n", annotation.getDescription());
+          out.printf("Position : %s\n", annotation.getBoundingPoly());
+        }
+      }
+    }
+  }
+
+  /**
+   * Detects text in the specified remote image.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: Google Cloud Storage

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 12]
```diff
@@ -0,0 +1,1271 @@
...
+              color.getColor().getRed(),
+              color.getColor().getGreen(),
+              color.getColor().getBlue());
+        }
+      }
+    }
+  }
+
+  /**
+   * Detects image properties such as color frequency from the specified remote image.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: as above, Google Cloud Storage

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 13]
```diff
@@ -0,0 +1,1271 @@
...
+            annotation.getViolence(),
+            annotation.getRacy());
+      }
+    }
+  }
+  // [END vision_detect_safe_search]
+
+  // [START vision_detect_safe_search_uri]
+  /**
+   * Detects whether the specified remote image has features you would want to moderate.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: remote image => image in a Google Cloud Storage bucket ?

File: vision/beta/cloud-client/src/test/java/com/example/vision/DetectIT.java [hunk 14]
```diff
@@ -0,0 +1,350 @@
+/*
+  Copyright 2016, Google, Inc.
```
Comment (jabubake) [inline vision/beta/cloud-client/src/test/java/com/example/vision/DetectIT.java]: s/2016/2017

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 15]
```diff
@@ -0,0 +1,1271 @@
...
+            pageText = pageText + blockText;
+          }
+        }
+        out.println("\nComplete annotation:");
+        out.println(annotation.getText());
+      }
+    }
+  }
+  // [END vision_detect_document_uri]
+}
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: new line at eof

File: vision/beta/cloud-client/src/test/java/com/example/vision/DetectIT.java [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+    String[] args = {"fulltext", "gs://" + BUCKET + "/vision/text.jpg"};
+    Detect.argsHelper(args, out);
+
+    // Assert
+    String got = bout.toString();
+    assertThat(got).contains("After preparation is complete, the ");
+    assertThat(got).contains("37%");
+    assertThat(got).contains("Word text: class (confidence:");
+  }
+}
```
Comment (jabubake) [inline vision/beta/cloud-client/src/test/java/com/example/vision/DetectIT.java]: new line EOF

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java [hunk 17]
```diff
@@ -0,0 +1,1271 @@
+/**
+ * Copyright 2017, Google, Inc.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
```
Comment (jabubake) [inline vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: Please use the same formatting for license as in the other files.
<p> tags don't seem to be required here.

File: vision/beta/cloud-client/pom.xml (hunk 2 above)
Comment (nnegrey) [reply vision/beta/cloud-client/pom.xml]: Removed

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java (hunk 17 above)
Comment (nnegrey) [reply vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: Updated

File: vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java (hunk 4 above)
Comment (nnegrey) [reply vision/beta/cloud-client/src/main/java/com/example/vision/Detect.java]: Updating here and in original file.

File: vision/beta/cloud-client/README.md (hunk 1 above)
Comment (nnegrey) [reply vision/beta/cloud-client/README.md]: Using mvn exec:java 
Removing the jar

File: vision/beta/cloud-client/README.md [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+### Analyze an image
+
+```
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/your-project-credentials.json
+```
+
+```
+mvn exec:java -DDetect -Dexec.args="logos './resources/logos.png'"
```
Comment (jabubake) [inline vision/beta/cloud-client/README.md]: Please provide more examples here.
given you now have faces, logos etc

**Pull request merged**